### PR TITLE
Machine Learning Questions Thing

### PR DESCRIPTION
**Pull request by ghostdevv**

### CODE REVIEW

File: src/events/question_button.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+		const { data: prediction, error } = await supabase
+			.from('predictions')
+			.select()
+			.eq('message_id', message_id)
+			.limit(1)
+			.single();
+
+		if (error)
+			return void interaction.followUp(
```
Comment (gtm-nayan) [inline src/events/question_button.ts:24]: Probably should await all of these other wise they won't get caught.

File: src/events/question_button.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+				wrap_in_embed('Unable to find that message prediction'),
+			);
+
+		if (interaction.user.id != prediction.author_id)
+			return void interaction.followUp(
+				wrap_in_embed('You can only report your own predictions'),
+			);
```
Comment (gtm-nayan) [inline src/events/question_button.ts:31]: Threadlords instead?

**Review by gtm-nayan — CHANGES_REQUESTED** [review]

File: src/ml/guess.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	const tensor = await encoder.embed([content]);
+
+	const predicted = model.predict(tensor);
+	const result = Array.isArray(predicted) ? predicted[0] : predicted; // ts thinks it's an array so lets pander
+
+	const chance = result.dataSync()[0];
+
+	return {
+		question: chance > 0.85,
+		message: chance < 0.85,
```
Comment (gtm-nayan) [inline src/ml/guess.ts:25]:
```suggestion
		message: chance <= 0.85,
```

File: src/events/question.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+				.setCustomId(`prediction_${message.id}`)
+				.setEmoji('<:far_snape:837108258484781126>');
+
+			const { error } = await supabase.from('predictions').insert({
+				message_id: message.id,
+				prediction: prediction.chance,
+				author_id: message.author.id,
+			});
+
+			message.reply({
```
Comment (gtm-nayan) [inline src/events/question.ts:35]:
```suggestion
			await message.reply({
```

File: src/events/question_button.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		await interaction.deferReply({ ephemeral: true });
+
+		const message_id = interaction.customId.slice('prediction_'.length);
+
+		const { error } = await supabase
+			.from('predictions')
+			.select()
+			.eq('message_id', message_id)
+			.limit(1)
+			.single();
```
Comment (gtm-nayan) [inline src/events/question_button.ts:22]:
```suggestion

```

Don't need this anymore right?

File: src/events/question_button.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+		if (error)
+			return void (await interaction.followUp(
```
Comment (gtm-nayan) [inline src/events/question_button.ts:25]: The brackets add a bit of noise, remove the void.

File: src/events/question_button.ts (hunk 5 above)
Comment (ghostdevv) [reply src/events/question_button.ts]: Moved the error check

File: src/events/question_button.ts (hunk 6 above)
Comment (ghostdevv) [reply src/events/question_button.ts]: Type errors :/

**Review by gtm-nayan — APPROVED** [review]

**Pull request merged**